packages/@react-spectrum/s2/src/SegmentedControl.tsx

```diff
@@ -0,0 +1,260 @@
+/*
+ * Copyright 2024 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {centerBaseline} from './CenterBaseline';
+import {ContextValue, DEFAULT_SLOT, Provider, TextContext as RACTextContext, Radio, RadioGroup, RadioGroupProps, RadioGroupStateContext, RadioProps} from 'react-aria-components';
+import {createContext, forwardRef, ReactNode, RefObject, useCallback, useContext, useRef} from 'react';
+import {DOMRef, DOMRefValue, FocusableRef} from '@react-types/shared';
+import {focusRing, getAllowedOverrides, StyleProps} from './style-utils' with {type: 'macro'};
+import {IconContext} from './Icon';
+import {pressScale} from './pressScale';
+import {size, style} from '../style/spectrum-theme' with {type: 'macro'};
+import {Text, TextContext} from './Content';
+import {useDOMRef, useFocusableRef} from '@react-spectrum/utils';
+import {useLayoutEffect} from '@react-aria/utils';
+import {useSpectrumContextProps} from './useSpectrumContextProps';
+
+export interface SegmentedControlProps extends Omit<RadioGroupProps, 'isReadOnly' | 'name' | 'isRequired' | 'isInvalid' | 'validate' | 'validationBehavior' | 'children' | 'className' | 'style' | 'aria-label' | 'orientation'>, StyleProps{
+  /**
+   * The content to display in the segmented control.
+   */
+  children: ReactNode,
+  /**
+   * Whether the segmented control is disabled.
+   */
+  isDisabled?: boolean,
+  /**
+   * Defines a string value that labels the current element.
+   */
+  'aria-label': string
+}
+export interface SegmentedControlItemProps extends Omit<RadioProps, 'children' | 'className' | 'style' | 'onHoverStart' | 'onHoverEnd' | 'onHoverChange'>, StyleProps {
+  /**
+   * The content to display in the control item.
+   */
+  children: ReactNode
+}
+
+export const SegmentedControlContext = createContext<ContextValue<SegmentedControlProps, DOMRefValue<HTMLDivElement>>>(null);
+
+const segmentedControl = style<{size: string}>({
+  font: 'control',
+  display: 'flex',
+  backgroundColor: 'gray-100',
+  borderRadius: 'lg',
+  width: 'full'
+}, getAllowedOverrides());
+
+const controlItem = style({
+  position: 'relative',
+  display: 'flex',
+  forcedColorAdjust: 'none',
+  color: {
+    default: 'gray-700',
+    isHovered: 'neutral-subdued',
+    isSelected: 'neutral',
+    isDisabled: 'disabled',
+    forcedColors: {
+      default: 'ButtonText',
+      isDisabled: 'GrayText',
+      isSelected: 'HighlightText'
+    }
+  },
```

Comment:
looks like the text in light mode doesn't show up in HCM, at least when using chrome's emulation
<img width="290" alt="image" src="https://github.com/user-attachments/assets/e3621e0f-573e-4d2b-b036-1936bdee0609">

Reply:
And when disabled as well.